flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/VertexParallelism.java

```diff
@@ -23,14 +23,27 @@
 import java.util.Map;
 
 /**
- * Core result of {@link SlotAllocator#determineParallelism(JobInformation, Collection)}, describing
- * the parallelism each vertex could be scheduled with.
- *
- * <p>{@link SlotAllocator} implementations may encode additional information to be used in {@link
- * SlotAllocator#tryReserveResources(VertexParallelism)}.
+ * Core result of {@link SlotAllocator#determineParallelism(JobInformation, Collection)} among with
+ * {@link org.apache.flink.runtime.scheduler.adaptive.JobSchedulingPlan.SlotAssignment
+ * slotAssignments}, describing the parallelism each vertex could be scheduled with.
  */
-public interface VertexParallelism {
-    Map<JobVertexID, Integer> getMaxParallelismForVertices();
+public class VertexParallelism {
```

Comment:
If this is not an interface, does this method signature in SlotAllocator still make sense?

```
    Optional<? extends VertexParallelism> determineParallelism(
            JobInformation jobInformation, Collection<? extends SlotInfo> slots);
```

This has also been removed from the contract:

```
 * <p>{@link SlotAllocator} implementations may encode additional information to be used in {@link
 * SlotAllocator#tryReserveResources(VertexParallelism)}.
 ```

Reply:
Good point, I'll adjust the interface and its javadoc.